ext/reflection/php_reflection.c

```diff
@@ -6041,7 +6041,7 @@ ZEND_METHOD(ReflectionProperty, getSettableType)
 	}
 
 	/* Get-only virtual property can never be written to. */
-	if ((prop->flags & ZEND_ACC_VIRTUAL) && !prop->hooks[ZEND_PROPERTY_HOOK_SET]) {
+	if (prop->hooks && !prop->hooks[ZEND_PROPERTY_HOOK_SET]) {
```

Comment:
This should not return never for get-only backed props. I think the virtual check should not be dropped.

Reply:
Ah right, I get it now why this was checked this way. I'm gonna put the hooks pointer check first though so that this if and the one below can be factored out commonly by the compiler.

Reply:
Yeah, I'm surprised this wasn't covered by a test. I'll add one soon.

Reply:
Added one here: https://github.com/php/php-src/commit/85fa983fe6df75fe6159c076f44748351b9cfe09